frontend/src/components/SideNavBar.tsx

```diff
@@ -1,15 +1,19 @@
+"use client";
+import { useState } from 'react';
 import React from 'react'
 import svgIcons from "../app/svgIcons";
 import SideNavBarButton from "@/components/SideNavBarButton";
 export default function SideNavBar() {
+  const [currentSelection, setCurrentSelection] = useState("Home");
+  const updatedText = currentSelection;
   return (
-    <div className="side-nav-bar">
-      <SideNavBarButton svgIcon={svgIcons.home} text="Home" />
-        <SideNavBarButton svgIcon={svgIcons.menuManager} text="Menu Manager" />
-        <SideNavBarButton svgIcon={svgIcons.orderTracking} text="Order Tracking" />
-        <SideNavBarButton svgIcon={svgIcons.inventory} text="Inventory Tracker" />
-        <SideNavBarButton svgIcon={svgIcons.productivity} text="Productivity Metrics" />
-        <SideNavBarButton svgIcon={svgIcons.settings} text="Settings" />
+    <div className="side-nav-bar">      
+        <SideNavBarButton svgIcon={svgIcons.home} currentText={updatedText} setCurrentSelection={setCurrentSelection} text="Home" />
```

Comment:
I wonder if, rather than passing down the logic to determine if the button is selected to the component, we just have an extra prop `selected` which is true if selected and false otherwise. Then you could do `selected={currentText === "Home"}` and you don't need to pass the selectedText around.

Reply:
I went ahead and lifted up the logic, and it worked, with a minor caveat:
For some reason, now I can't automatically change the color of the home button when the page loads for the first time. it stays unchanged no matter how many times I change it, even when the initial value of the state is "Home." It also doesn't allow me to select it as it's already selected. It's some pretty odd behavior. It can be selected if I choose another option first. So I decided to make the initial state of the sidebar component "none" to  prevent that from happening.